src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Remote/IRemoteServiceProvider.cs

```diff
@@ -1,13 +1,14 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
+using System;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.ExternalAccess.Razor;
 
 namespace Microsoft.CodeAnalysis.Razor.Remote;
 
-internal interface IRemoteClientProvider
+internal interface IRemoteServiceProvider
 {
-    Task<RazorRemoteHostClient?> TryGetClientAsync(CancellationToken cancellationToken);
+    ValueTask<TResult?> TryInvokeAsync<TService, TResult>(Solution solution, Func<TService, RazorPinnedSolutionInfoWrapper, CancellationToken, ValueTask<TResult>> invocation, CancellationToken cancellationToken) where TService : class;
```

Comment:
Is this expected to complete synchronously most of the time? If not, should it be a `Task`?

Reply:
It's ValueTask all the way down to the `TryInvokeAsync` methods in  https://github.com/dotnet/roslyn/blob/main/src/Workspaces/Core/Portable/Remote/RemoteServiceConnection.cs#L22 where this call ultimately ends up, so I just followed the pattern. I'm guessing it ultimately comes from service hub, but even if its Roslyn, it feels to me like we should trust they picked ValueTask for a reason. Thoughts?

Reply:
Going to merge this to unblock myself, but will happily follow up.

Reply:
If it is VT all the way down this makes sense to me